06-walkthrough-1.Rmd

```diff
@@ -271,89 +304,228 @@ Note that they're now combined, even though the course data has many more rows:
+It looks like *a lot* of gradebook items were repeated - likely across the
+different sections of the same course (we would be curious to hear what you find
+if you investigate this!).
 
-<!-- This is really trivial and obvious; need a new/better relationship -->
+Let's use what we just did, but to find the unique values at the student-level.
+Thus, instead of exploring unique gradebook items, we will explore unique
+students (still accounting for the course, as students could enroll in more than
+one course.) This time, we will add the `keep_all = TRUE` argument.
 
-```{r}
-ggplot(dat, aes(x = TimeSpent, y = percentage_earned)) +
-    geom_point()
+``` {r}
+dat <- distinct(dat, course_id, student_id, .keep_all = TRUE)
 ```
 
-There appears to be *some* relationship. What if we added a line of best fit - a linear model?
+This is a much smaller data frame - with one row for each sudnet in the course
+(instead of the 29,701 rows which we would be interested in were we analyzing
+this data at the level of specific students' grades for specific gradebook
+items). Now that our data are ready to go, we can start to ask some questions of
+the data,
 
-```{r}
+## Visualizations and Models
+
+### The relationship between time spent on course and percentage of points earned
+
+One thing we might be wondering is how time spent on course is related to
+students' final grade.
+
+We note that ggplot2, which we use to create these plots, is discussed further in chapter XXX.
+
+``` {r}
+ggplot(dat, aes(x = TimeSpent, y = percentage_earned)) + # this tells ggplot2 what variables to map to what feature of a plot, here, x- anx y-axis locations
+  geom_point() # creates a point with x- and y-axis coordinates specified above
+```
+
+There appears to be *some* relationship. What if we added a line of best fit - a
+linear model?
+
+``` {r}
 ggplot(dat, aes(x = TimeSpent, y = percentage_earned)) +
-    geom_point() + 
-    geom_smooth(method = "lm")
+  geom_point() + # same as above
+  geom_smooth(method = "lm") # this adds a line of best fit; method = "lm" tells ggplot2 to make the line linear, whereas the default is a smooth like
 ```
 
-So, it appeares that the more time students spent on the course, the more points they earned.
+So, it appeares that the more time students spent on the course, the more points
+they earned.
 
-# Linear model (regression)
+## Linear model (regression)
 
-We can find out exactly what the relationship is using a linear model.
+We can find out exactly what the relationship is using a linear model. We also
+discuss linear models in walkthrough XXX.
 
-```{r}
+Here, we predict `percentage_earned`, or the percentage of the total points that
+are possible for a student to earn. Here, percentage earned is the dependent, or
+*y*-variable, and so we enter it first, after the `lm()` command, before the
+tilde (`~`) symbol. To the right of the tilde is one independent variable,
```

Comment:
I'm really seeing the value of @jmostipak 's spreadsheet here. I have a very similar explanation of `lm` in the longitudinal data walkthrough. I'll start an issue and milestone for taking inventory of this